File: solutions/401. Binary Watch/401-2.py
class Solution:
  def readBinaryWatch(self, turnedOn: int) -> list[str]:
    ans = []

    for h in range(12):
      for m in range(60):
        if h.bit_count() + m.bit_count() == turnedOn:
          ans.append(f'{h}:{m:02d}')

    return ans

File: solutions/401. Binary Watch/401-2.java
class Solution {
  public List<String> readBinaryWatch(int turnedOn) {
    List<String> ans = new LinkedList<>();

    for (int h = 0; h < 12; ++h)
      for (int m = 0; m < 60; ++m)
        if (Integer.bitCount(h) + Integer.bitCount(m) == turnedOn)
          ans.add(h + (m < 10 ? ":0" : ":") + m);

    return ans;
  }
}
